proto/topsql_agent.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package topsql_agent;
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+import "google/protobuf/empty.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+// TopSQLAgent is the persistent agent service for TopSQL records
+service TopSQLAgent {
+  // CollectTikv is called periodically (e.g. per minute) to save the in-memory TopSQL records
+  rpc CollectTikv(stream CPUTimeRequestTikv) returns (google.protobuf.Empty) {}
+}
+
+message CPUTimeRequestTikv {
+  repeated uint64 timestamp_list = 1;  // timestamp in second
+  repeated uint32 cpu_time_ms_list = 2;  // this can be greater than 1000 when counting concurrent running SQL queries
+  bytes payload = 3;  // the specific encoding is opaque to tikv-server
```

Comment:
What is payload? Is it ResourceTag?

Reply:
Yes, I should name it more specifically.